src/modules/src/p2pDTR/queueing.c

```diff
@@ -0,0 +1,144 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *
+ * queueing.c
+ *
+ *  Created on: 14.02.2021
+ *      Author: Christos Zosimidis
+ *
+ *  Modified for the P2P protocol by: Bitcraze AB
+ * 
+ */
+
+
+#include "queueing.h"
+
+#include "debug.h"
+
+// TX SRV packet queue
+static xQueueHandle TX_DATA_queue;
+
+// RX SRV packet queue 
+static xQueueHandle RX_SRV_queue;
+
+// RX DATA packet queue
+static xQueueHandle RX_DATA_queue;
+
+
+static xQueueHandle *getQueueHandler(DTRQueue_Names qName){
+	switch(qName){
+		case TX_DATA_Q:
+			return &TX_DATA_queue;
+		case RX_SRV_Q:
+			return &RX_SRV_queue;
+		case RX_DATA_Q:
+			return &RX_DATA_queue;
+		default:
+			// The q enum is not valid
+			DEBUG_PRINT("Invalid DTR queue name\n");
+			ASSERT(0);
+			return NULL;
+	}
+}
+
+
+void DTRqueueingInit(){
+	// TX SRV queue
+	STATIC_MEM_QUEUE_ALLOC(TX_DATA_queue, TX_DATA_QUEUE_SIZE, sizeof(DTRpacket));
+	TX_DATA_queue = STATIC_MEM_QUEUE_CREATE(TX_DATA_queue);
+	DEBUG_QUEUE_MONITOR_REGISTER(TX_DATA_queue);
+
+	// RX SRV queue
+	STATIC_MEM_QUEUE_ALLOC(RX_SRV_queue, RX_SRV_QUEUE_SIZE, sizeof(DTRpacket));
+	RX_SRV_queue = STATIC_MEM_QUEUE_CREATE(RX_SRV_queue);
+	DEBUG_QUEUE_MONITOR_REGISTER(RX_SRV_queue);
+
+	// RX DATA queue
+	STATIC_MEM_QUEUE_ALLOC(RX_DATA_queue, RX_DATA_QUEUE_SIZE, sizeof(DTRpacket));
+	RX_DATA_queue = STATIC_MEM_QUEUE_CREATE(RX_DATA_queue);
+	DEBUG_QUEUE_MONITOR_REGISTER(RX_DATA_queue);
+
+}
+
+uint8_t DTRgetNumberOfPacketsInQueue(DTRQueue_Names qName){
+	return (uint8_t) uxQueueMessagesWaiting(*getQueueHandler(qName));
+}
+
+bool DTRisPacketInQueueAvailable(DTRQueue_Names qName) {
+	return uxQueueMessagesWaiting(*getQueueHandler(qName)) > 0;
+}
+
+bool DTRgetPacketFromQueue(DTRpacket *packet, DTRQueue_Names qName, uint32_t timeout){
+	// notice that xQueuePeek is used instead of xQueueReceive, because the packet is not removed from the queue
+    //TODO: make a separate function for this
+	bool received_success;
+	switch (qName)
+	{
+	case TX_DATA_Q:
+		received_success = xQueuePeek(TX_DATA_queue, packet, timeout) == pdTRUE;
+		break;
+	
+	case RX_SRV_Q:
+		received_success = xQueueReceive(RX_SRV_queue, packet, timeout) == pdTRUE;
+		break;
+	case RX_DATA_Q:
+		received_success = xQueueReceive(RX_DATA_queue, packet, timeout) == pdTRUE;
+		break;
+
+	default:
+		DEBUG_PRINT("Invalid DTR queue name\n");
+		received_success = false;
+		break;
+	}
+	return received_success;
+}
+
+bool DTRreceivePacketWaitUntil(DTRpacket *packet, DTRQueue_Names qName, uint32_t timeout_ms, bool *new_packet_received){
+	*new_packet_received = xQueueReceive(*getQueueHandler(qName), packet, M2T(timeout_ms)) == pdTRUE;
+	return true;
+}
+
+bool DTRinsertPacketToQueue(DTRpacket *packet, DTRQueue_Names qName) {
+	bool res = xQueueSend(*getQueueHandler(qName),(void *) packet, 0) == pdTRUE;
```

Comment:
the return value from `xQueueSend()` is technically `BaseType_t` and not bool. It is possibly clearer to do
`if (pdTRUE != res) ...`, not sure

Reply:
It was already like this but I will add `( ... )` to make it more visible